File: SDKs/asana.com/1.0/go/pkg/models/shared/webhookresponse.go
package shared

import (
	"time"
)

// WebhookResponseFilters
// A set of filters to specify a whitelist for what types of events will be delivered.
type WebhookResponseFilters struct {
	Action          *string  `json:"action,omitempty"`
	Fields          []string `json:"fields,omitempty"`
	ResourceSubtype *string  `json:"resource_subtype,omitempty"`
	ResourceType    *string  `json:"resource_type,omitempty"`
}

// WebhookResponse
// Webhook objects represent the state of an active subscription for a server to be updated with information from Asana. This schema represents the subscription itself, not the objects that are sent to the server. For information on those please refer to the [Event](/docs/tocS_Event) schema.
type WebhookResponse struct {
	Active             *bool                    `json:"active,omitempty"`
	CreatedAt          *time.Time               `json:"created_at,omitempty"`
	Filters            []WebhookResponseFilters `json:"filters,omitempty"`
	Gid                *string                  `json:"gid,omitempty"`
	LastFailureAt      *time.Time               `json:"last_failure_at,omitempty"`
	LastFailureContent *string                  `json:"last_failure_content,omitempty"`
	LastSuccessAt      *time.Time               `json:"last_success_at,omitempty"`
	Resource           *AsanaNamedResource      `json:"resource,omitempty"`
	ResourceType       *string                  `json:"resource_type,omitempty"`
	Target             *string                  `json:"target,omitempty"`
}

File: SDKs/asana.com/1.0/java/lib/src/main/java/openapisdk/models/shared/WebhookResponse.java
package openapisdk.models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import openapisdk.utils.DateTimeSerializer;
import openapisdk.utils.DateTimeDeserializer;

/**
 * WebhookResponse
 * Webhook objects represent the state of an active subscription for a server to be updated with information from Asana. This schema represents the subscription itself, not the objects that are sent to the server. For information on those please refer to the [Event](/docs/tocS_Event) schema.
**/
public class WebhookResponse {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("active")
    public Boolean active;
    public WebhookResponse withActive(Boolean active) {
        this.active = active;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("created_at")
    public OffsetDateTime createdAt;
    public WebhookResponse withCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("filters")
    public WebhookResponseFilters[] filters;
    public WebhookResponse withFilters(WebhookResponseFilters[] filters) {
        this.filters = filters;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("gid")
    public String gid;
    public WebhookResponse withGid(String gid) {
        this.gid = gid;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("last_failure_at")
    public OffsetDateTime lastFailureAt;
    public WebhookResponse withLastFailureAt(OffsetDateTime lastFailureAt) {
        this.lastFailureAt = lastFailureAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("last_failure_content")
    public String lastFailureContent;
    public WebhookResponse withLastFailureContent(String lastFailureContent) {
        this.lastFailureContent = lastFailureContent;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("last_success_at")
    public OffsetDateTime lastSuccessAt;
    public WebhookResponse withLastSuccessAt(OffsetDateTime lastSuccessAt) {
        this.lastSuccessAt = lastSuccessAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("resource")
    public AsanaNamedResource resource;
    public WebhookResponse withResource(AsanaNamedResource resource) {
        this.resource = resource;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("resource_type")
    public String resourceType;
    public WebhookResponse withResourceType(String resourceType) {
        this.resourceType = resourceType;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("target")
    public String target;
    public WebhookResponse withTarget(String target) {
        this.target = target;
        return this;
    }
}